#!/bin/bash

#Assembles the reads to generate a consensus sequence

set -eu

#Running velvet

mkdir -p results/velvet_out

echo Assembling Hcamelina_female_genome using velvet. Recommended Version 1.2.10.

velveth results/velvet_out 41 \
-fastq.gz -shortPaired \
-separate \
results/clean_reads/S3-Hcamelina-F.R1_val_1.fq.gz \
results/clean_reads/S3-Hcamelina-F.R2_val_2.fq.gz && \
velvetg results/velvet_out/ \
-exp_cov auto \
-cov_cutoff auto \
-ins_length 149 
# Remove intermediate files 

rm results/velvet_out/{Graph2,LastGraph,Log,PreGraph,Roadmaps,Sequences} 

#Rename the generated assembly, contigs.fa to match the organism,hcamelina

mv results/velvet_out/contigs.fa results/velvet_out/hcamelina_f_genome.fa 

#Rename the stats file to match the organism

mv results/velvet_out/stats.txt results/velvet_out/hcamelina_f_genome_stats.txt

echo Completed assembling hcamellina_male genome with Velvet successfully !!!!


# 81: This sets up the Velvet hash table and k-mer size. The 81 argument specifies the k-mer size (i.e., the length of each sequence fragment used to assemble the genome).
#-fastq.gz -shortPaired -separate: These options specify that the input reads are in FASTQ format (-fastq.gz), that they are paired-end (-shortPaired), and that the paired reads are in separate files (-separate).
# &&: This is a command separator that allows you to chain multiple commands together in a single line.
# velvetg results/velvet_out/: This runs the Velvet genome assembly algorithm on the hash table generated by velveth.
# results/velvet_out/ directory is the output directory where the assembled contigs will be written.
#-exp_cov auto -cov_cutoff auto -ins_length 149: These options specify various parameters for the assembly. -exp_cov and -cov_cutoff are used to estimate the expected coverage and coverage cutoff for the contigs, respectively. In this case, the auto option is used to let Velvet automatically estimate these values. -ins_length specifies the insert length of the paired-end reads, which is used to help 
# Velvet estimate the distance between the paired read





