% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\#Set working dir
setwd(``/home/kebaso/Documents/projects/hippo/hvariegata\_female/results/orthofinder/Results\_Apr21/Comparative\_Genomics\_Statistics'')

\#load all packages required
pacman::p\_load(magrittr,knitr,kableExtra,reshape2,tidyverse,ggplot2,vcfR)

\#plot/visualize in a table form the overall statistics
knitr::kable(read.table(``Statistics\_Overall.tsv'',fill = T, nrows =
24, sep = ``\t")) \%\textgreater\% kable\_styling(bootstrap\_options =
c(''striped'', ``hover'', ``condensed'', ``responsive''))

\#section 2: Genes per orthogroups genes\_per\_OG \textless-
read.delim(``Statistics\_Overall.tsv'',fill = T, skip = 25, nrows = 20,
sep = ``\t") kable(genes\_per\_OG) \%\textgreater\%
kable\_styling(bootstrap\_options = c(''striped'', ``hover'',
``condensed'', ``responsive''))

\#plot the results instead of a table genes\_per\_OG.m \textless-
melt(genes\_per\_OG, id.vars =
``Average.number.of.genes.per.species.in.orthogroup'', factorsAsStrings
= T) \#genes\_per\_OG.m
genes\_per\_OG.m\(Average.number.of.genes.per.species.in.orthogroup <- factor(genes_per_OG.m\)Average.number.of.genes.per.species.in.orthogroup,levels
=
as.character(genes\_per\_OG\$Average.number.of.genes.per.species.in.orthogroup))
\#plot ggplot(genes\_per\_OG.m,
aes(x=Average.number.of.genes.per.species.in.orthogroup , y=value,
fill=variable)) + geom\_bar(stat=``identity'') +
facet\_grid(variable\textasciitilde., scales = ``free\_y'') +
scale\_fill\_viridis\_d() + theme\_bw() + theme(legend.position =
``none'')

\#section 3 line 47-54:species per orthogroups species\_per\_OG
\textless- read.delim(``Statistics\_Overall.tsv'',fill = T, skip = 46,
nrows = 7, sep = ``\t", header = T) kable(species\_per\_OG)
\%\textgreater\% kable\_styling(bootstrap\_options = c(''striped'',
``hover'', ``condensed'', ``responsive''))

\#for easy interpretation,plot the table ggplot(species\_per\_OG,
aes(x=as.factor(Number.of.species.in.orthogroup),
y=Number.of.orthogroups,
fill=as.factor(Number.of.species.in.orthogroup))) +
geom\_bar(stat=``identity'') + scale\_fill\_viridis\_d() + theme\_bw() +
theme(legend.position = ``none'') + xlab(``Number of species on
orthogroup'') + ylab(``Number of orthogroups'')

\#*****************************Statistics
PerSpecies************************\# \#*File used
Statistics\_PerSpecies\#

\#read the file--get only the first 10 rows overall\_stats \textless-
read.table(``Statistics\_PerSpecies.tsv'',fill = T, nrows = 10, sep =
``\t", header = T)

\#get species names species\_names \textless-
colnames(overall\_stats){[}-1{]} \#species\_names

\#plot a table with this information knitr::kable(overall\_stats)
\%\textgreater\% kable\_styling(bootstrap\_options = c(``striped'',
``hover'', ``condensed'', ``responsive''))

\#As these results are very numerous, lets graph them. We will do a
subset of the non-percentage data first: overall\_stats.number
\textless- overall\_stats{[}grep(x = overall\_stats\$X, pattern =
``Number''),{]} \#overall\_stats.number \#using X as id variables
overall\_stats.number \textless- melt(overall\_stats.number)
\#overall\_stats.number

\#plot out this result/in a facet graph,first format X by doing away
with `Number of'
overall\_stats.number\(X <- gsub(overall_stats.number\)X, pattern =
``Number of'', replacement = ``\,``) \%\textgreater\%
tools::toTitleCase() \#overall\_stats.number

ggplot(overall\_stats.number, aes(x=variable , y=value, fill=X)) +
geom\_bar(stat=``identity'') + facet\_grid(X\textasciitilde., scales =
``free\_y'') + scale\_fill\_viridis\_d() + theme\_bw() +
theme(legend.position = ``none'') + ylab(``Number'')

\#plot \% more standardized instead of numerical value
overall\_stats.percentage \textless- overall\_stats{[}grep(x =
overall\_stats\(X, pattern = "Number", invert = T),] #overall_stats.percentage #using X as id variables overall_stats.percentage <- melt(overall_stats.percentage) #overall_stats.percentage #plot % data #format the pattern name by replacing "Percentage of" with "" overall_stats.percentage
\)X \textless- gsub(overall\_stats.percentage\$X, pattern = ``Percentage
of'', replacement = ``\,``) \%\textgreater\% tools::toTitleCase()

ggplot(overall\_stats.percentage, aes(x=variable , y=value, fill=X)) +
geom\_bar(stat=``identity'') + facet\_grid(X\textasciitilde., scales =
``free\_y'') + scale\_fill\_viridis\_d() + theme\_bw() +
theme(legend.position = ``none'') + ylab(``Percentage (\%)'')

\#per species per orthologs statistics \#line 35-56 perecentage\_stats
\textless- read.delim(``Statistics\_PerSpecies.tsv'',fill = T, skip =
35, nrows = 20, sep = ``\t", header = T) \#perecentage\_stats
colnames(perecentage\_stats){[}-1{]} \textless- species\_names
\#perecentage\_stats

\#tabular representation knitr::kable(perecentage\_stats)
\%\textgreater\% kable\_styling(bootstrap\_options = c(``striped'',
``hover'', ``condensed'', ``responsive''))

\#melt the tabular data
perecentage\_stats\(Number.of.genes.per.species.in.orthogroup <- factor(perecentage_stats\)Number.of.genes.per.species.in.orthogroup,
levels = perecentage\_stats\$Number.of.genes.per.species.in.orthogroup)

\#using number.of.genes.per.species.in.orthogroup as id variables
perecentage\_stats \textless- melt(perecentage\_stats)
\#perecentage\_stats

\#plot \% stat ggplot(perecentage\_stats,
aes(x=Number.of.genes.per.species.in.orthogroup , y=value,
fill=variable)) + geom\_bar(stat=``identity'', position = ``dodge'') +
scale\_fill\_viridis\_d() + theme\_bw() + theme(legend.position =
``bottom'') + ylab(``Percentage (\%)'')

\#************************OrthogroupsGeneCountfile;shows us the count of
genes in each orthogroups***************\# \#*files for this part of
work is Orthogroups.GeneCount.tsv----gives a detailed look of all above
work;ordered by count genes

setwd(``/home/kebaso/Documents/projects/hippo/hvariegata\_female/results/orthofinder/Results\_Apr21/Orthogroups'')

og.gene\_count \textless- read.table(``Orthogroups.GeneCount.tsv'',
header = T) og.gene\_count head(og.gene\_count) \%\textgreater\%
knitr::kable() \%\textgreater\% kable\_styling(bootstrap\_options =
c(``striped'', ``hover'', ``condensed'', ``responsive'')) \#\#most genes
in the head are attributed to OG0000000 whose contributed by A.nidulans

\#We can observe the general pattern of the orthogroups by \#plotting a
heat map for the first 100 orthogroups without the Total column;use vcfR
for heatmap.bp OG.mtrix \textless-
as.matrix(og.gene\_count{[}c(1:100),-c(1,8){]}) rownames(OG.mtrix)
\textless- og.gene\_count{[}c(1:100),1{]} OG.mtrix{[}OG.mtrix == 0{]}
\textless- NA heatmap.bp(OG.mtrix)

\#We should be able to see the contributions of each \#species per OG by
removing all OG with values of 0 and counting the remaining OG:
non\_zero\_OG \textless- og.gene\_count{[}apply(og.gene\_count{[},-1{]},
1, function (x) sum(x == 0) == 0),{]} nrow(non\_zero\_OG)

\#look at first 10 head(non\_zero\_OG, n=10) \%\textgreater\%
knitr::kable() \%\textgreater\% kable\_styling(bootstrap\_options =
c(``striped'', ``hover'', ``condensed'', ``responsive''))

\#look at the last 10 rows tail(non\_zero\_OG, n=10) \%\textgreater\%
knitr::kable() \%\textgreater\% kable\_styling(bootstrap\_options =
c(``striped'', ``hover'', ``condensed'', ``responsive''))

\end{document}
