#Set working dir
setwd("/home/kebaso/Documents/projects/hippo/hvariegata_female/results/orthofinder/Results_Apr21/Comparative_Genomics_Statistics")

#load all packages required
pacman::p_load(magrittr,knitr,kableExtra,reshape2,tidyverse,ggplot2,vcfR)

#plot/visualize in a table form the overall statistics
knitr::kable(read.table("Statistics_Overall.tsv",fill = T, nrows = 24, sep = "\t")) %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))


#section 2: Genes per orthogroups
genes_per_OG <- read.delim("Statistics_Overall.tsv",fill = T, skip = 25, nrows = 20, sep = "\t")
kable(genes_per_OG) %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))

#plot the results instead of a table
genes_per_OG.m <- melt(genes_per_OG, id.vars = "Average.number.of.genes.per.species.in.orthogroup", factorsAsStrings = T)
#genes_per_OG.m
genes_per_OG.m$Average.number.of.genes.per.species.in.orthogroup <- factor(genes_per_OG.m$Average.number.of.genes.per.species.in.orthogroup,levels = as.character(genes_per_OG$Average.number.of.genes.per.species.in.orthogroup))
#plot
ggplot(genes_per_OG.m, aes(x=Average.number.of.genes.per.species.in.orthogroup , y=value, fill=variable)) + 
  geom_bar(stat="identity") + facet_grid(variable~., scales = "free_y") + 
  scale_fill_viridis_d() + theme_bw() + theme(legend.position = "none") 

#section 3 line 47-54:species per orthogroups
species_per_OG <- read.delim("Statistics_Overall.tsv",fill = T, skip = 46, nrows = 7, sep = "\t", header = T)
kable(species_per_OG) %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))

#for easy interpretation,plot the table
ggplot(species_per_OG, aes(x=as.factor(Number.of.species.in.orthogroup), y=Number.of.orthogroups, fill=as.factor(Number.of.species.in.orthogroup))) + 
  geom_bar(stat="identity") + scale_fill_viridis_d() + theme_bw() + 
  theme(legend.position = "none")  + 
  xlab("Number of species on orthogroup") + 
  ylab("Number of orthogroups")


#*****************************Statistics PerSpecies************************#
#*File used Statistics_PerSpecies#

#read the file--get only the first 10 rows
overall_stats <- read.table("Statistics_PerSpecies.tsv",fill = T, nrows = 10, sep = "\t", header = T)

#get species names
species_names <- colnames(overall_stats)[-1]
#species_names

#plot a table  with this information
knitr::kable(overall_stats) %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))


#As these results are very numerous, lets graph them. We will do a subset of the non-percentage data first:
overall_stats.number <- overall_stats[grep(x = overall_stats$X, pattern = "Number"),]
#overall_stats.number
#using X as id variables
overall_stats.number <- melt(overall_stats.number)
#overall_stats.number

#plot out this result/in a facet graph,first format X by doing away with 'Number of'
overall_stats.number$X <- gsub(overall_stats.number$X, pattern = "Number of ", replacement = "") %>% tools::toTitleCase()
#overall_stats.number

ggplot(overall_stats.number, aes(x=variable , y=value, fill=X)) + 
  geom_bar(stat="identity") + facet_grid(X~., scales = "free_y") + 
  scale_fill_viridis_d() + theme_bw() + theme(legend.position = "none") + ylab("Number")

#plot % more standardized instead of numerical value
overall_stats.percentage <- overall_stats[grep(x = overall_stats$X, pattern = "Number", invert = T),]
#overall_stats.percentage
#using X as id variables
overall_stats.percentage <- melt(overall_stats.percentage)
#overall_stats.percentage
#plot % data
#format the pattern name by replacing "Percentage of" with ""
overall_stats.percentage$X <- gsub(overall_stats.percentage$X, pattern = "Percentage of ", replacement = "") %>% tools::toTitleCase()

ggplot(overall_stats.percentage, aes(x=variable , y=value, fill=X)) + 
  geom_bar(stat="identity") + facet_grid(X~., scales = "free_y") + 
  scale_fill_viridis_d() + theme_bw() + theme(legend.position = "none") + ylab("Percentage (%)")

#per species per orthologs statistics
#line 35-56
perecentage_stats <- read.delim("Statistics_PerSpecies.tsv",fill = T, skip = 35, nrows = 20, sep = "\t", header = T)
#perecentage_stats
colnames(perecentage_stats)[-1] <- species_names
#perecentage_stats

#tabular representation
knitr::kable(perecentage_stats) %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))

#melt the tabular data
perecentage_stats$Number.of.genes.per.species.in.orthogroup <- factor(perecentage_stats$Number.of.genes.per.species.in.orthogroup, levels = perecentage_stats$Number.of.genes.per.species.in.orthogroup)

#using number.of.genes.per.species.in.orthogroup as id variables
perecentage_stats <- melt(perecentage_stats)
#perecentage_stats

#plot % stat
ggplot(perecentage_stats, aes(x=Number.of.genes.per.species.in.orthogroup , y=value, fill=variable)) + 
  geom_bar(stat="identity", position = "dodge") + scale_fill_viridis_d() + 
  theme_bw() + theme(legend.position = "bottom") + ylab("Percentage (%)")



#************************OrthogroupsGeneCountfile;shows us the count of genes in each orthogroups***************#
#*files for this part of work is Orthogroups.GeneCount.tsv----gives a detailed look of all above work;ordered by count genes

setwd("/home/kebaso/Documents/projects/hippo/hvariegata_female/results/orthofinder/Results_Apr21/Orthogroups")

og.gene_count <- read.table("Orthogroups.GeneCount.tsv", header = T)
og.gene_count
head(og.gene_count) %>% knitr::kable() %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
##most genes in the head are attributed to OG0000000 whose contributed by A.nidulans

#We can observe the general pattern of the orthogroups by 
#plotting a heat map for the first 100 orthogroups without the Total column;use vcfR for heatmap.bp
OG.mtrix <- as.matrix(og.gene_count[c(1:100),-c(1,8)])
rownames(OG.mtrix) <- og.gene_count[c(1:100),1]
OG.mtrix[OG.mtrix == 0] <- NA
heatmap.bp(OG.mtrix)

#We should be able to see the contributions of each 
#species per OG by removing all OG with values of 0 and counting the remaining OG:
non_zero_OG <- og.gene_count[apply(og.gene_count[,-1], 1, function (x) sum(x == 0) == 0),]
nrow(non_zero_OG)

#look at first 10
head(non_zero_OG, n=10) %>% knitr::kable() %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))

#look at the last 10 rows
tail(non_zero_OG, n=10) %>% knitr::kable() %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))






